components/Chat/TextMessagePart.tsx

```diff
@@ -1,16 +1,12 @@
 "use client";
 
 import { TextPart } from "@/types/reasoning";
-import ChatMarkdown from "./ChatMarkdown";
+import Markdown from "./ChatMarkdown-v2/markdown";
 
 interface TextMessagePartProps {
   part: TextPart;
 }
 
 export function TextMessagePart({ part }: TextMessagePartProps) {
-  return (
-    <ChatMarkdown>
-      {part.text}
-    </ChatMarkdown>
-  );
+  return <Markdown content={part.text} />
```

Comment:
if your goal is simply `mermaid chart`
why are you fundamentally changing how we render markdown in the app?

currently, we're following vercel Chat SDK.

Why are you changing how we render `<Markdown />` in a pull request titled only for `mermaid charts`?

Reply:
I created a new Markdown component using the `marked` library and placed it in the ChatMarkdown-v2 directory. Earlier ChatMarkdown component is still there.

Reply:
Why are you changing the markdown component used?
Why not `update` the existing markdown to support the functionality you need to add?